### PR TITLE
[WIP] Replace Appium and IntegrationTests with Detox

### PR DESCRIPTION
**Pull request by grabbou**
This is WIP that aims to replace Appium with Detox. Note that it intentionally targets `0.52-branch` as that's the one I am testing stuff at (I need real CircleCI setup as well).

Reasoning:
Right now, we use Appium to validate that Dev Tools work properly on Android. The setup is a bit hacky (not part of `package.json`) and is Android-specific. Since this speeds up releases a bit and makes sure nothing gets omitted, I decided to take it step forward and make a cross-platform test suite to verify all the things relevant from the release perspective.

Since we already have Detox integrated, I decided to place `__e2e__` folder (my own convention to match Jest) within `RNTester` to validate its screens as well. 

My goal is to get rid of `IntegrationTests` folder and unify all the integration testing solutions to just one. Then, we can think of running that on many devices, with AWS DeviceFarm as a possible candidate.

I am discussing few ideas that I have in mind with Wix developers. I am open to any ideas, suggestions, and improvements as I keep working on this PR.

### CODE REVIEW

**Comment by grabbou** [comment]
This is WIP at this stage and not meant for reviews. The architecture (setup) is done - feel free to comment regarding that part.

Next step - test existing Appium things, create a separate `circleci` step and clean things up.

**Comment by grabbou** [comment]
Right now, I am block on this https://github.com/wix/detox/issues/551 and will continue working inside Detox repo to move this PR forward.

Meanwhile, ideas are welcome!

**Comment by orta** [comment]
Very cool

**Comment by grabbou** [comment]
It's okay Facebook bot - I am waiting for Detox people to merge few PRs and
will get back working on this PR.

**Comment by grabbou** [comment]
Finally started working on it again after a while - thanks to Detox team, we now have support for `shake()` which makes it possible to test dev features.

Right now, I've added a simple test case for Chrome Debugger - will keep working on that with plan to land it this week.

File: detox/setup.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const detox = require('detox');
+const config = require('../package.json').detox;
+
+// Set the default timeout
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
```
Comment (thymikee) [inline detox/setup.js]: You can use `jest.setTimeout(120000)` now instead.

File: RNTester/__e2e__/devTools.spec.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+   * Test whether Chrome Debugger works by turning it on
+   * and seeing whether it refreshes the app correctly
+   */
+  it('should have `Debug In Chrome` working', async () => {
+    await device.shake();
+    await element(by.text('Debug JS Remotely')).tap();
+    await expect(element(by.text('RNTester'))).toBeVisible();
+    await device.shake();
+    await element(by.text('Stop Remote JS Debugging')).tap();
+    await waitFor(element(by.text('RNTester'))).toBeVisible().withTimeout(2000);
```
Comment (rotemmiz) [inline RNTester/__e2e__/devTools.spec.js]: why did you use `waitFor` here ? does it have problems syncing ?

Comment (rotemmiz) [reply RNTester/__e2e__/devTools.spec.js]: anyway, if you do use `waitFor` (and you shouldn't), it must be followed by an expectation.
`await expect(element(by.text('RNTester'))).toBeVisible();`

**Comment by cpojer** [comment]
I'm assuming this PR is dead. We are now using detox for tests, so should be all good?

**Comment by grabbou** [comment]
Either way, lack of interest here suggests this is not a priority now.

**Comment by rotemmiz** [comment]
1. @d4vidi is working on delivering Detox Android as a precompiled aar. This should solve the integration issue of Detox Android into RNTester. We'll update.
2. Circle CI have recently added Android x86 emulator support, this means we can test Detox Android in CI.

**Comment by d4vidi** [comment]
True that -- indeed this is a top priority task.
What's holding me back now, however (not in this context specifically, but rather in moving forward with Android), is this RN issue I've just submitted: https://github.com/facebook/react-native/issues/23870

**Pull request closed without merging**